packages/camera/example/lib/main.dart

```diff
@@ -24,149 +26,316 @@ IconData cameraLensIcon(CameraLensDirection direction) {
-      columnChildren.add(
-        new Text('Camera error ${controller.value.errorDescription}'),
-      );
-    } else {
-      columnChildren.add(
-        new Expanded(
-          child: new Padding(
-            padding: const EdgeInsets.all(5.0),
-            child: new Center(
-              child: new AspectRatio(
-                aspectRatio: controller.value.aspectRatio,
-                child: new CameraPreview(controller),
-              ),
-            ),
-          ),
-        ),
-      );
+    if (mounted) {
+      setState(() {});
     }
-    return new Scaffold(
-      appBar: new AppBar(
-        title: const Text('Camera example'),
-      ),
-      body: new Column(children: columnChildren),
-      floatingActionButton: (controller == null)
-          ? null
-          : new FloatingActionButton(
-              child: const Icon(Icons.camera),
-              onPressed: controller.value.isStarted ? capture : null,
-            ),
-    );
   }
 
-  Widget imageWidget() {
-    return new Expanded(
-      child: new Align(
-        alignment: Alignment.centerRight,
-        child: new SizedBox(
-          child: new Image.file(new File(imagePath)),
-          width: 64.0,
-          height: 64.0,
-        ),
-      ),
-    );
+  void onTakePictureButtonPressed() {
+    takePicture().then((String filePath) {
+      if (mounted) {
+        setState(() {
+          imagePath = filePath;
+          videoController?.dispose();
+          videoController = null;
+        });
+        showInSnackBar('Picture saved to $filePath');
+      }
+    });
   }
 
-  Widget playPauseButton() {
-    return new FlatButton(
-      onPressed: () {
-        setState(
-          () {
-            if (controller.value.isStarted) {
-              controller.stop();
-            } else {
-              controller.start();
-            }
-          },
-        );
-      },
-      child:
-          new Icon(controller.value.isStarted ? Icons.pause : Icons.play_arrow),
-    );
+  void onVideoRecordButtonPressed() {
+    startVideoRecording().then((String filePath) {
+      if (mounted) setState(() {});
+      showInSnackBar('Saving video to $filePath');
+    });
   }
 
-  Future<Null> capture() async {
-    if (controller.value.isStarted) {
-      final Directory tempDir = await getTemporaryDirectory();
-      if (!mounted) {
-        return;
+  Future<void> onStopButtonPressed() async {
+    await stopVideoRecording();
+    if (!mounted) {
+      return null;
+    }
+    showInSnackBar('Video recorded to: $videoPath');
+  }
+
+  Future<String> startVideoRecording() async {
+    if (!controller.value.isInitialized) {
+      showInSnackBar('Error: select a camera first.');
+      return null;
+    }
+    final Directory extDir = await getApplicationDocumentsDirectory();
+    final String dirPath = '${extDir.path}/Movies/flutter_test';
+    await new Directory(dirPath).create(recursive: true);
+    final String filePath = '$dirPath/${timestamp()}.mp4';
```

Comment:
One problem with timestamping instead of overwriting the previous file is that the user has no way to delete pictures. It might not be so important for an example app. Just a thought.

Reply:
We could improve this indeed, but I suggest doing it in another PR. Files can be deleted in Android by clearing the cache (not ideal but then it's just an example app).